dali/base/dadfs.cpp

```diff
@@ -4921,6 +4920,24 @@ StringBuffer &CDistributedFilePart::getStorageFilePath(StringBuffer & path, unsi
     return path;
 }
 
+unsigned CDistributedFilePart::getStripeNum(unsigned copy)
+{
+    unsigned nc = copyClusterNum(copy, nullptr);
+    IClusterInfo &cluster = parent.clusters.item(nc);
+    const char *planeName = cluster.queryGroupName();
+    if (isEmptyString(planeName))
+    {
+        StringBuffer lname;
+        parent.getLogicalName(lname);
+        throw new CDFS_Exception(DFSERR_EmptyStoragePlane, lname.str());
+    }
+    Owned<IStoragePlane> storagePlane = getDataStoragePlane(planeName, false);
+    if (!storagePlane)
+        throw new CDFS_Exception(DFSERR_MissingStoragePlane, planeName);
+
+    return calcStripeNumber(partIndex, parent.lfnHash, storagePlane->numDevices());
```

Comment:
minor: it is probably not a big efficiency concern, but getStripeNum will probably be called repeatedly (e.g. when a process is dealing with multiple parts), it might be better to precalculate the {part#, copy#}->stripe# mapping in the parent once, to avoid each call looking up the data plane etc.